apps/examples/testcase/le_tc/filesystem/fs_main.c

```diff
@@ -1662,6 +1662,71 @@ static void libc_stdio_fread_tc(void)
 }
 
 /**
+* @testcase         libc_stdio_freopen_tc
+* @brief            Open file by freopen
+* @scenario         Open file
+* @apicovered       freopen
+* @precondition     NA
+* @postcondition    NA
+*/
+static void libc_stdio_freopen_tc(void)
+{
+	FILE *fp;
+	char *filename = VFS_FILE_PATH;
+
+	fp = freopen(NULL, "w", NULL);
+	TC_ASSERT_EQ("freopen", fp, NULL);
+
+	fp = freopen(filename, "w", NULL);
+	TC_ASSERT("freopen", fp);
+	fclose(fp);
+
+	fp = freopen(filename, "w", fp);
+	TC_ASSERT("freopen", fp);
+	fclose(fp);
+
+	fp = freopen(filename, "r+", fp);
+	TC_ASSERT("freopen", fp);
+	fclose(fp);
+
+	fp = freopen(filename, "rb", fp);
+	TC_ASSERT("freopen", fp);
+	fclose(fp);
+
+	fp = freopen(filename, "rx", fp);
+	TC_ASSERT("freopen", fp);
+	fclose(fp);
+
+	/* Nagative cases with invalid mode. It will return NULL */
+
+	fp = freopen(NULL, "b", fp);
+	TC_ASSERT_EQ("freopen", fp, NULL);
+
```

Comment:
These macros for negative case should be TC_ASSERT_EQ_CLEANUP("freopen", fp, NULL, fclose(fp)).
If fp is not NULL, fp may point something wrong in this case.
So fp should be cleaned before return

Reply:
@jeongarmy In case of invalid arguments, freopen will always return NULL as per code.
Hence no clean should be required for these cases.

Reply:
Right. In this case, freopen should be return NULL.
But if fp is not NULL because of any reason, it should be cleaned like negative testcases in libc_stdio_fopen_tc
if (fp != NULL) {... (it is needed to free fp) ...}

Reply:
For negative cases it should return NULL. Why do you think for any reason it will return otherwise ?? I dont understand.
According to the current implementation of fopen also, we dont need to use TC_ASSERT_EQ_CLEANUP as it always returs NULL in case of negative TC's

Reply:
I'm not saying it can return non-NULL in any cases.
We are making testcases so we should focus on input and output only, not internal implementation.
When we make tc, we check returned value or something from function.
And if results are wrong, we should do somthing for cleaning it regarding that function
Although this function always returns NULL with invalid parameter, we should add this logic for cleaning logically